src/engine/contactsengine.cpp

```diff
@@ -515,13 +529,27 @@ class JobThread : public QThread
         {
             QMutexLocker locker(&m_mutex);
             m_running = false;
-            m_wait.wakeOne();
         }
+        m_wait.wakeOne();
         wait();
     }
 
     void run();
 
+    bool waitForRunning()
+    {
+        QMutexLocker locker(&m_mutex);
+        if (!m_running) {
+            m_wait.wait(&m_mutex);
+        }
```

Comment:
Dtor wakes with ```m_running == false``` but it is not checked here. Isn't there a danger of race condition?